bait-files/src/main/java/com/google/sps/servlets/RequestServlet.java

```diff
@@ -24,21 +31,74 @@
   private static final String PICTURE = "picture-input";
   private static final String PHONE = "phone-input";
   private static final String NOTES = "notes-input";
+  private static final String API_ERROR = "The request to this API failed.";
+  private static HttpClient client;
+
+  public RequestServlet() {
+    this.client = HttpClient.newHttpClient();
+  }
+
+  public RequestServlet(HttpClient client) {
+    this.client = client;
+  }
+
+  // Wrapper class for Request object and API result JsonObjects
+  private class RequestWrapper {
+    Request request;
+    JsonObject phoneResults;
+    JsonObject emailResults;
+
+    private RequestWrapper(Request request, JsonObject phoneResults, JsonObject emailResults) {
+      this.request = request;
+      this.phoneResults = phoneResults;
+      this.emailResults = emailResults;
+    }
+  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     try {
       long requestId = Long.parseLong(request.getParameter("requestId"));
       Request userRequest = ObjectifyService.ofy().load().type(Request.class).id(requestId).now();
+
       if (userRequest == null) {
         HashMap<String, String> responseParameters = new HashMap<String, String>();
         responseParameters.put("redirect", "true");
         response.setContentType("application/json;");
         response.getWriter().println(new Gson().toJson(responseParameters));
         return;
       }
+
+      JsonObject phoneResults = new JsonObject();
+
+      Url phoneUrl =
+          ObjectifyService.ofy().load().type(Url.class).filter("name", "phone-api").first().now();
+      if (phoneUrl != null && !userRequest.phoneNum.isEmpty()) {
+        try {
+          phoneResults = JsonParser.parseString(doGetAPI(phoneUrl.url + userRequest.phoneNum))
+                             .getAsJsonObject();
+        } catch (InterruptedException e) {
+          phoneResults =
+              JsonParser.parseString("{\"results_unavailable\": true}").getAsJsonObject();
+        }
+      }
```

Comment:
You could extract this chunk of stuff to a private method:
private JsonObject getPhoneApiResults() { ... }